components/Mediator.vue

```diff
@@ -0,0 +1,282 @@
+        @load-hints="$event.waitUntil(loadHints())">
+        <template slot="header">
+          <div style="padding-right: 10px; margin-bottom: 5px">
+            <strong>{{relyingDomain}}</strong> wants credentials
+          </div>
+          <div v-if="loading">
+            Loading options... <i class="fa fa-circle-o-notch fa-spin"></i>
+          </div>
+          <div v-else-if="!needsStorageAccess">
+            <div v-if="hintOptions.length === 0">
+              <div class="wrm-heading">Warning</div>
+              <div>
+                You don't have the credentials requested by this website.
+                Please check <strong>{{relyingDomain}}</strong> to find out
+                how to obtain the credentials you need to continue.
+              </div>
+            </div>
+            <div v-else class="wrm-heading">
+              Choose an option to continue
+            </div>
+          </div>
+        </template>
+      </wrm-hint-chooser>
+    </div>
+
+    <div v-else-if="display === 'credentialStore'">
+      <wrm-hint-chooser
+        :hints="hintOptions"
+        default-hint-icon="fa-user"
+        :confirm-button="needsStorageAccess"
+        @confirm="selectHint"
+        @cancel="cancel()"
+        @load-hints="$event.waitUntil(loadHints())">
+        <template slot="header">
+          <div style="padding-right: 10px; margin-bottom: 5px">
+            <strong>{{relyingDomain}}</strong> wants to store credentials
+          </div>
+          <div v-if="loading">
+            Loading options... <i class="fa fa-circle-o-notch fa-spin"></i>
+          </div>
+          <div v-else-if="!needsStorageAccess">
+            <div v-if="hintOptions.length === 0">
+              You don't have a digital wallet to store credentials. Please
+              visit a digital wallet website to install one.
+            </div>
+            <div v-else class="wrm-heading">
+              Choose an option to continue
+            </div>
+          </div>
+        </template>
+      </wrm-hint-chooser>
+    </div>
+  </div>
+</template>
+<script>
+/*!
+ * New BSD License (3-clause)
+ * Copyright (c) 2017-2018, Digital Bazaar, Inc.
+ * All rights reserved.
+ */
+/* global navigator */
+'use strict';
+
+import * as polyfill from 'credential-mediator-polyfill';
+import {utils} from 'web-request-rpc';
+import HandlerWindowHeader from './HandlerWindowHeader.vue';
+import Vue from 'vue';
+
+let deferredCredentialOperation;
+let resolvePermissionRequest;
+
+export default {
+  name: 'Mediator',
+  async created() {
+    if(window.location.ancestorOrigins &&
+      window.location.ancestorOrigins.length > 0) {
+      this.relyingOrigin = window.location.ancestorOrigins[0];
+    } else {
+      const {origin} = this.$route.query;
+      this.relyingOrigin = origin;
+    }
+
+    this.relyingDomain = utils.parseUrl(this.relyingOrigin).hostname;
+
+    // TODO: is this the appropriate place to run this?
+    loadPolyfill(this);
```

Comment:
In another conversation it was suggested there's no need to await on result of this and might want to make created() not async.